src/types.ts

```diff
@@ -22,13 +22,45 @@ export interface SdDigestHashmap {
   [sd_digest: string]: Disclosure;
 }
 
+export interface DisclosureClaim {
+  key?: string;
+  value: any;
+}
+
+export type ArrayDisclosureFrame = Array<boolean | DisclosureFrame>;
+
+export type DisclosureFrame = {
+  [key: string]: DisclosureFrame | ArrayDisclosureFrame | unknown;
```

Comment:
@linasi typescript was saying it's an error otherwise, that _sd and _self doesn't match [key: string]'s type
Do you know of a better way to define this type/data structure?

Reply:
You can add these keys separately.
```
export type DisclosureFrame = {
  _sd: string[];
  _self: boolean;
  [key: string]: DisclosureFrame | ArrayDisclosureFrame;
}
```